src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -1896,20 +1896,25 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_RequestSEDActiv
 
     if (!onOff && delayIdle && CHIP_DEVICE_CONFIG_SED_ACTIVE_THRESHOLD.count() != 0)
     {
-        err = DeviceLayer::SystemLayer().StartTimer(CHIP_DEVICE_CONFIG_SED_ACTIVE_THRESHOLD, RequestSEDModeChange, this);
+        err = DeviceLayer::SystemLayer().StartTimer(CHIP_DEVICE_CONFIG_SED_ACTIVE_THRESHOLD, RequestSEDModeUpdate, this);
         if (CHIP_NO_ERROR == err)
         {
+            if (!mDelayIdleTimerRunning)
```

Comment:
Why do we start a timer even if mDelayIdleTimerRunning, only to then only increment consumers if the timer is not already set to running? Would it make sense to `ReturnErrorCodeIf(mDelayIdleTimerRunning, CHIP_NO_ERROR)` ?

Reply:
Looking at the docs, I see `If called with @a aComplete and @a aAppState identical to an existing timer,
     *       the currently-running timer will first be cancelled.`
     
 So the intent here is to move the timer execution point. Maybe a comment would help. Reading StartTimer documentation is not always obvious.

Reply:
I believe we want to restart the timer. Let's say we have 3 consumers:
1. Consumer 1 switches to idle with delay: the timer is started
2. Consumer 2 switches to idle with delay: the timer is restarted
3. Consumer 3 switches to idle without delay: just decrease the counter and wait for the timer from step 2 to actually switch the device to idle.

Reply:
Damian comprehension is indeed correct, will add a comment in a follow up PR.